spacy_llm/tasks/span/parser.py

```diff
@@ -0,0 +1,67 @@
+from typing import Callable, Dict, Iterable, Tuple
+
+from spacy.tokens import Span
+
+from ...tasks.util import find_substrings
+
+
+def _format_response(
+    response: str, normalizer: Callable[[str], str], label_dict: Dict[str, str]
+) -> Iterable[Tuple[str, Iterable[str]]]:
+    """Parse raw string response into a structured format.
+    response (str): LLM response.
+    normalizer (Callable[[str], str]): normalizer function.
+    label_dict (Dict[str, str]): Mapping of normalized to non-normalized labels.
+    RETURNS (Iterable[Tuple[str, Iterable[str]]]): Formatted response.
+    """
+    output = []
+    assert normalizer is not None
+
+    for line in response.strip().split("\n"):
+        # Check if the formatting we want exists
+        # <entity label>: ent1, ent2
+        if line and ":" in line:
+            label, phrases = line.split(":", 1)
+            norm_label = normalizer(label)
+            if norm_label in label_dict:
+                # Get the phrases / spans for each entity
+                if phrases.strip():
+                    _phrases = [p.strip() for p in phrases.strip().split(",")]
+                    output.append((label_dict[norm_label], _phrases))
+
+    return output
+
+
+def parse_responses(responses: Iterable[str], **kwargs) -> Iterable[Span]:
```

Comment:
I don't think `kwargs` makes sense here. It actually loses quite a bit of clarity in the function. I'm assuming this is done so we have a single `TaskResponseParserProtocol`.

If you're looking at this function in isolation, we don't validate `kwargs` at all, we just assume these keys are present. This "works" but is not very easy to understand and is a source of potential bugs.

Reply:
I agree on the merits of explicitly stating arguments over `kwargs`. I originally went for the latter because I couldn't find another way to make the types of `_parse_responses` agree with the parent's `__init__()` signature. I'll have another look into how to at least approximate a typing match there.

Reply:
Check out `ParamSpec`, that's the general way to type a callable and say inherit these arguments

Reply:
Good point, thanks. Not sure if after transitioning to explicit signatures for the concrete `parse_responses()` functions using `ParamSpec` over `Callable[..., Iterable[Any]]` makes a difference though, will look into it.